carddrawer/src/main/java/com/meli/android/carddrawer/model/CardDrawerView.java

```diff
@@ -298,20 +308,53 @@ private void show(@NonNull final GenericPaymentMethod genericPaymentMethod) {
         showTag(genericPaymentMethod, genericTagText, genericFrontLayout);
 
         genericPaymentMethod.setPaymentMethodImage(paymentMethodImage);
+
         genericTitle.setText(genericPaymentMethod.getTitle().getText());
         genericTitle.setTextColor(genericPaymentMethod.getTitle().getColor());
-        showGenericPaymentSubtitle(genericPaymentMethod.getSubtitle());
-        frontBackground.setColorFilter(genericPaymentMethod.getBackgroundColor(), PorterDuff.Mode.SRC_ATOP);
-        backBackground.setColorFilter(genericPaymentMethod.getBackgroundColor(), PorterDuff.Mode.SRC_ATOP);
+        genericTitle.setTypeface(genericTitle.getTypeface(), CardDrawerFont.from(genericPaymentMethod.getTitle().getWeight()).getStyle());
+
+        showGenericText(genericPaymentMethod);
+
+        applyBackground(frontBackground, backBackground, genericPaymentMethod);
     }
 
-    protected void showGenericPaymentSubtitle(@Nullable final GenericPaymentMethod.Text subtitle) {
+    private void applyBackground(@NotNull final AppCompatImageView frontBackground,
+                                 @NotNull final AppCompatImageView backBackground,
+                                 @NotNull final GenericPaymentMethod genericPaymentMethod
+    ) {
+        final List<String> gradientColors = genericPaymentMethod.getGradientColors();
+        final int backgroundColor = genericPaymentMethod.getBackgroundColor();
+
+        if (gradientColors != null) {
+            final GradientDrawable gradientDrawable = ViewHelper.getGradientDrawable(getContext(), gradientColors);
+            frontBackground.setImageDrawable(gradientDrawable);
+            backBackground.setImageDrawable(gradientDrawable);
+        } else {
+            frontBackground.getBackground().setColorFilter(backgroundColor, PorterDuff.Mode.SRC_ATOP);
+            backBackground.getBackground().setColorFilter(backgroundColor, PorterDuff.Mode.SRC_ATOP);
+        }
+    }
+
+    protected void showGenericText(@NotNull final GenericPaymentMethod genericPaymentMethod) {
+        final GenericPaymentMethod.Text description = genericPaymentMethod.getDescription();
+        final GenericPaymentMethod.Text subtitle = genericPaymentMethod.getSubtitle();
+
+        if (description != null) {
```

Comment:
This "if-else" looks similar to the one for subtitle, maybe we can avoid that duplication?

Reply:
Yes, my idea was to abstract since there is duplicate code, but I'm going to break inheritance haha

Reply:
mmm why? maybe we can just extract this to a method that receives an AppCompatTextView and a GenericPaymentMethod.Text, or also we can do a kotlin extension for AppCompatTextView for this 🤔